### PR TITLE
Issue1985 clean pumps

### LINKED ISSUE #1985: Clean up pump control
This issue is to clean up pumps control sequences as a preparation for the open loop review.

### PR DESCRIPTION
**Pull request by JayHuLBL**
This closes #1985.

### CODE REVIEW

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Pumps/ChilledWater/Controller.mo [hunk 1]
```diff
@@ -49,64 +49,73 @@ block Controller
   final parameter Integer pumInd[nPum]={i for i in 1:nPum}
     "Pump index, {1,2,...,n}";
 
-  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput uPumLeaLag[nPum]
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput uPumLeaLag[nPum] if
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Pumps/ChilledWater/Controller.mo:52]: It is a bit unclear to me what this input represents. Can you please clarify, as we now have lead and lag as categories, and the values are integers. Is it that the first value in the array is the index of the lead device?

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Pumps/ChilledWater/Controller.mo [hunk 2]
```diff
@@ -149,97 +158,114 @@ block Controller
...
 
 protected
-  Buildings.Controls.OBC.CDL.Integers.Sources.Constant conInt(final k=1)
+  Buildings.Controls.OBC.CDL.Integers.Sources.Constant conInt(final k=1) if
+    isHeadered
     "Constant one"
     annotation (Placement(transformation(extent={{-220,190},{-200,210}})));
-  Buildings.Controls.OBC.CDL.Routing.IntegerReplicator intRep(final nout=nPum)
+  Buildings.Controls.OBC.CDL.Routing.IntegerReplicator intRep(final nout=nPum) if
+    isHeadered
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Pumps/ChilledWater/Controller.mo:191]: maybe have final start in the new line for these?

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Pumps/ChilledWater/Subsequences/Validation/EnableLead_dedicated.mo [hunk 3]
```diff
@@ -11,38 +11,50 @@ model EnableLead_dedicated
     "Disable lead pump due to that the lead chiller has been proven off"
     annotation (Placement(transformation(extent={{40,-30},{60,-10}})));
 
+  Buildings.Controls.OBC.CDL.Logical.TrueHoldWithReset leaChiProOn(duration=
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Pumps/ChilledWater/Subsequences/Validation/EnableLead_dedicated.mo:14]: final?

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Pumps/ChilledWater/Validation/Controller.mo [hunk 4]
```diff
@@ -2,144 +2,146 @@ within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Pumps.ChilledWat
 model Controller "Validate chiller water pump control sequence"
 
   Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Pumps.ChilledWater.Controller
-    heaNoLoc(nPum=3, nPum_nominal=3)
+    heaNoLoc(
+    nPum=3,
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Pumps/ChilledWater/Validation/Controller.mo:6]: What is the latest consensus on using `final` keyword in validation tests? I thought for these, as they validate our implementation we are still supposed to use them.

**Comment by milicag** [comment]
@JayHuLBL Is in line 61 in the condenser water pump controller the variable `yLeaPum` a setpoint, that is an enabling status?

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Pumps/ChilledWater/Controller.mo (hunk 1 above)
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Pumps/ChilledWater/Controller.mo:52]: The array should be `[lead pump index, 1st lag pump index, 2nd lag pump index, ...]`

**Comment by JayHuLBL** [comment]
@milicag Addressed your comments
I will address the comments from Brandon today in separate issue.

**Pull request merged**